#Scripts used to analyze linked selection and demography in maize

Below is a description of each script used in Beissinger et

## Part 1: SFS, Genotype likelihoods, and SNP calling

**fullScript_genic.sh** Computes SFS, 2D-SFS, and diversity info for maize and teosinte in genic regions only.

**fullScript_intergenic.sh** Computes SFS, 2D-SFS, and diversity info for maize and teosinte in intergenic regions only.

**fullScript_WholeGenome.sh** Computes SFS, 2D-SFS, and diversity info for maize and teosinte, genome-wide.

**callSNPsAngsd_maize_July.sh** Calls SNPs in maize.

**callSNPsAngsd_teo_july.sh** Calls SNPs in teosinte.

**countSingletonsMaize.R** Counts maize singletons in 1000 bp windows

**countSingletonsTeo.R** Counts teosinte singletons in 1000 bp windows

**countSingletonsMaize_downsample.R** Downsamples maize to a sample size equal to teosinte and then counts maize singletons in 1000 bp windows, 

## Part 2: Code to study demography



## Part 2: R code to study selection


